vitest.workspace.ts

```diff
@@ -0,0 +1,82 @@
+import react from '@vitejs/plugin-react';
+import { defineWorkspace, WorkspaceProjectConfiguration } from 'vitest/config';
+
+const packages = [
+  'x-charts',
+  'x-charts-pro',
+  'x-date-pickers',
+  'x-date-pickers-pro',
+  'x-data-grid',
+  'x-data-grid-pro',
+  'x-data-grid-premium',
+  'x-tree-view',
+  'x-tree-view-pro',
+];
+
+const jsdomOnlyPackages = [
+  'x-license',
+  'x-internals',
+  'x-codemod',
+  'x-charts-vendor',
+  'eslint-plugin-material-ui',
+];
+
+const allPackages = [...packages, ...jsdomOnlyPackages];
+
+// Ideally we move the configuration to each package.
+// Currently it doesn't work because vitest doesn't detect two different configurations in the same package.
+// We could bypass this limitation by having a folder per configuration. Eg: `packages/x-charts/browser` & `packages/x-charts/jsdom`.
+
+export default defineWorkspace([
+  ...allPackages.flatMap((name): WorkspaceProjectConfiguration[] => [
+    // TODO: Decide on
+    // Manually changing the process.env in browser tests doesn't work.
+    // And alternative is to use `const {NODE_ENV} = process.env` in the code instead.
+    // x-license relies on `process.env.NODE_ENV` to determine the environment.
+    // x-internals has a weird chai import issue I couldn't fix
+    ...(jsdomOnlyPackages.includes(name)
```

Comment:
It feels suboptimial to me to have to define this on the workspace level and not on the individual package level.
Would it be possible to give each vitest project their own config file which extends from one or more base files?

Reply:
For instance we could just define workspaces as

```tsx
// ./vitest.workspace.ts
export default [
  './packages/*/vitest.config.{browser|jsdom}.ts'
]
```

and in the projects:

```tsx
// ./packages/x-data-grid/vitest.config.browser.ts
import { mergeConfig } from 'vitest/config'
import browserDefault from '@mui/test-utils/browserDefault'
export default mergeConfig(browserDefault, {
  // ...
})

// ./packages/x-data-grid/vitest.config.jsdom.ts
import jsdomDefault from '@mui/test-utils/jsdomDefault'
export default jsdomDefault
```

and omit the browser file for projects that are jsdom only

Not sure whether that works the way I expect it to be

Reply:
This suggestion looks spot on.
If it's possible, I'm also very much for it. 👍

Reply:
I think the feature to have two configs side by side was added recently, will test that.